test/Performance/run-perf_tests.py

```diff
@@ -0,0 +1,792 @@
+    team_id = get_id( conn, "TeamId", "Teams", "TeamName", team_name)
+    if team_id is None:
+        insert_team_entry( conn, team_name )
+        team_id = get_id( conn, "TeamId", "Teams", "TeamName", team_name)
+    return team_id
+ 
+def get_test_id( conn, test_name ):
+    """
+    This module retrieves the id of a Test entry. If the given test does not exists in PerformanceTests table, 
+    the module inserts it into the PerformanceTests table and retrieves its id.
+    Args:
+        conn (obj): Connection object to the Results database
+        test_name (str): The name of the test that the id is requested for
+    Returns:
+        id of the test
+    """
+    test_id = get_id( conn, "TestId", "PerformanceTests", "TestName", test_name )
+    if test_id is None:
+        insert_test_entry( conn, test_name )
+        test_id = get_id( conn, "TestId", "PerformanceTests", "TestName", test_name )
+    return test_id
+ 
+def get_driver_id( conn, driver_name ):
+    """
+    This module retrieves the id of a Driver entry. If the given driver does not exists in Drivers table, 
+    the module inserts it into the Drivers table and retrieves its id.
+    Args:
+        conn (obj): Connection object to the Results database
+        driver_name (str): The name of the driver that the id is requested for
+    Returns:
+        id of the driver
+    """
+    driver_path = get_path_to_driver( driver_name )
+    driver_hash = get_sha1_file( driver_path )
+    driver_id = get_id_no_quote( conn, "DriverId", "Drivers", "SHA1", driver_hash )
+    if driver_id is None:
+        insert_driver_entry( conn, driver_path, driver_hash )
+        driver_id = get_id_no_quote( conn, "DriverId", "Drivers", "SHA1", driver_hash )
+    return driver_id
+       
+def insert_result_entry_and_get_id( conn, test_id, client_id, driver_id, server_id, team_id, success ):
+    """
+    This module inserts a new result entry into PerformanceResults table and retrieves its id.
+    Args:
+        conn (obj): Connection object to the Results database
+        test_id (int): The id of the test
+        client_id (int): The id of the client that the test was run on
+        driver_id (int): The id of the driver that the test was run against
+        server_id (int): The id of the server that the test was run against
+        team_id (int): The id of the team that the test belongs to
+        success (int): 0 if the test failed, 1 otherwise
+    Returns:
+        id of the result
+    """
+    query = "INSERT INTO PerformanceResults( TestId, ClientId, DriverId, ServerId, TeamId, Success ) OUTPUT INSERTED.ResultId VALUES( {0}, {1}, {2}, {3}, {4}, {5} )"
+    cursor = conn.cursor()
+    cursor.execute( query.format( test_id, client_id, driver_id, server_id, team_id, success ))
+    result_id = cursor.fetchone()
+    cursor.close()
+    if result_id is not None:
+        return result_id[0]
+    return id
+ 
+def insert_key_value( conn, table_name, result_id, key, value ):

```

Comment:
suggest changing the name of this function to something like insert_result_key_value(). insert_key_value() sounds very generic (i.e., sounds like it can be used to insert something into the server table, or the client table).